Core/Src/spi_command_handler.c

```diff
@@ -5,170 +5,199 @@
 #include <string.h>
 
 extern SPI_HandleTypeDef hspi1;
-//		spi_receive(&rx_data, 1);
-static uint8_t cmd;
+extern uint8_t cam_to_nand_transfer_flag;
 
-uint8_t * get_image_buffer();
+uint32_t image_length = 0x5F3; // Only here for testing purposes
+static uint32_t count = 0x0FFF0000;
 
+uint8_t ack = 0xAA;
+uint8_t nack = 0x0F;
+uint8_t transmit_ack;
+
+/**
+ * @brief
+ * 		Transmit data of given size over SPI bus in blocking mode
+ *
+ * @param
+ * 		*tx_data: pointer to transmit data
+ * 		data_length: numbers of bytes to be sent
+ */
 void spi_transmit(uint8_t *tx_data, uint16_t data_length) {
-	HAL_SPI_Transmit(&hspi1, tx_data, data_length, HAL_MAX_DELAY);
+    HAL_SPI_Transmit(&hspi1, tx_data, data_length, HAL_MAX_DELAY);
 }
 
-void spi_receive(uint8_t *rx_data, uint16_t data_length) {
-	HAL_SPI_Receive(&hspi1, rx_data, data_length, HAL_MAX_DELAY);
+/**
+ * @brief
+ * 		Receive data of given size over SPI bus in interrupt mode
+ * @param
+ * 		*rx_data: pointer to receive data
+ * 		data_length: numbers of bytes to be receive
+ */
+void spi_receive(uint8_t *rx_data, uint16_t data_length) { HAL_SPI_Receive_IT(&hspi1, rx_data, data_length); }
+
+/**
+ * @brief
+ * 		Verifies if command from OBC is valid or not
+ * @param
+ * 		obc_cmd: Command from OBC
+ * @return
+ * 		1 if valid command, 0 if not
+ */
+int spi_verify_command(uint8_t obc_cmd) {
```

Comment:
add case for watchdog timer ack. We could probably get rid of handle_wdt() in command_handler.c and just use this.

Reply:
For sure, is the watchdog time command from the OBC a sort of a ping? That will periodically check if iris is alive?

Reply:
Yeah, it will just ping Iris over spi and look for an ack